File: Rasha_Toine/Rasha/QLForm/src/ast/expression/UnaryExpression.java
package ast.expression;

public class UnaryExpression extends Expression {

	private Expression expr;

	public UnaryExpression(Expression exp){
		this.setExpression(exp);
	}

	public Expression getExpression() {
		return expr;
	}

	public void setExpression(Expression expr) {
		this.expr = expr;
	}
}

File: DogeQL/src/main/kotlin/ql/ast/node/expression/UnaryExpression.kt
package ql.ast.node.expression

import ql.ast.location.SourceLocation
import ql.ast.node.expression.operation.UnaryOperation
import ql.visitor.QuestionnaireASTBaseVisitor

class UnaryExpression(val next: Expression, val operation: UnaryOperation, sourceLocation: SourceLocation)
    : Expression(sourceLocation) {

    override fun <T> accept(visitor: QuestionnaireASTBaseVisitor<T>): T {
        return visitor.visit(this)
    }

}